### PR TITLE
Random updates and corrections around ping360 integration 

### PR DESCRIPTION
**Pull request by patrickelectric**
- Add dynamic background
image
- Add controlPanel
image
- Improve PingTextField

### CODE REVIEW

File: src/sensor/ping.cpp [hunk 1]
```diff
@@ -552,6 +552,17 @@ void Ping::resetSensorLocalVariables()
...
 }
 
+QQuickItem* Ping::controlPanel(QObject *parent)
+{
+    QQmlEngine *engine = qmlEngine(parent);
+    if(!engine) {
+        qCDebug(PING_PROTOCOL_PING) << "No qml engine to load visualization.";
+        return nullptr;
+    }
+    QQmlComponent *component = new QQmlComponent(engine, QUrl("qrc:/Ping1DControlPanel.qml"));
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]: How is this deleted? Does engine own it and do the clean up?

File: qml/Ping360ControlPanel.qml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                            ping.sample_period = parseInt(text)*40
+                        }
+                    }
+                }
+
+                RowLayout {
+                     PingTextField {
+                        id: transmitFrequency
+                        title: "Transmit Frequency (kHz):"
+                        text: ping.transmit_frequency
```
Comment (jaxxzer) [inline qml/Ping360ControlPanel.qml]: `[15:42:11.200] default[Warning]: Ping360ControlPanel.qml(94) qrc:/Ping360ControlPanel.qml:94:25: Unable to assign [undefined] to QString`

File: src/sensor/ping.cpp (hunk 1 above)
Comment (patrickelectric) [reply src/sensor/ping.cpp]: Added smart pointer to remove any possible memory leak, the parent should take of it, but better safe than sorry.

File: qml/Ping360ControlPanel.qml (hunk 2 above)
Comment (patrickelectric) [reply qml/Ping360ControlPanel.qml]: Fixed.

**Pull request merged**